sweep/txgenerator.go

```diff
@@ -210,6 +212,16 @@ func createSweepTx(inputs []input.Input, outputPkScript []byte,
 		totalInput += btcutil.Amount(o.SignDesc().Output.Value)
 	}
 
+	// Add the outputs given, if any.
+	for _, o := range outputs {
+		sweepTx.AddTxOut(o)
+		requiredOutput += btcutil.Amount(o.Value)
+	}
+
+	if requiredOutput+txFee > totalInput {
```

Comment:
Style nit: commit body could've explained that this'll be used to ensure that if someone does a `sweepall` transaction, then we'd _force_ an extra UTXO to remain in the wallet to ensure that we could anchor down transactions if needed.